libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -69,6 +75,30 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
     // @Description: This selects which barometer will be the primary if multiple barometers are found
     // @Values: 0:FirstBaro,1:2ndBaro,2:3rdBaro
     AP_GROUPINFO("PRIMARY", 6, AP_Baro, _primary_baro, 0),
+    
+    // @Param: ADJ_RATE
+    // @DisplayName: baro adjustment rate
+    // @Description: Rate of barometer adjustment using GPS. If enabled (>=0.1), the barometer ground pressure is slowly adjusted during flight segments with constant altitude target by using GPS information.  Recommended value: 1cm/sec
+    // @Units: cm/second
+    // @Range: 0 10
+    // @Increment: 0.1
+    AP_GROUPINFO("ADJ_RATE", 7, AP_Baro, _adj_step, ADJ_RATE_DEFAULT),
+
+    // @Param: ADJ_TC
+    // @DisplayName: GPS filter time constant
+    // @Description: Time constant for the low pass filter applied to the GPS signal. As we only want to compensate for long-term trends, the should be fairly large (30sec...several mintes)
+    // @Units: seconds
+    // @Range: 10 600
+    // @Increment: 1
+    AP_GROUPINFO("ADJ_TC", 8, AP_Baro, _adj_timeconstant, ADJ_TC_DEFAULT),
+
+    // @Param: ADJ_DELAY
+    // @DisplayName: Alt adjust delay
+    // @Description: Dead time before alt tuning starts after constant alt target has been reached.
+    // @Units: seconds
+    // @Range: 10 600
+    // @Increment: 1
+    AP_GROUPINFO("ADJ_DELAY", 9, AP_Baro, _adj_delay, ADJ_DELAY_DEFAULT),
```

Comment:
The delay starts counting once a certain alt target has been reached, not after boot or takeoff.
That means that after Auto, Loiter, PosCtl or whatever mode with alt control has settled in a flight phase with constant altitude, the delay timer starts counting. After the delay  expired, the altitude difference seen in the GPS alt is used to calculate the correction. If the alt target is changed before the delay is over, no correction is done.